mono/metadata/w32process-unix.c

```diff
@@ -911,82 +911,64 @@ mono_w32process_try_get_modules (gpointer handle, gpointer *modules, guint32 siz
 	return TRUE;
 }
 
-guint32
-mono_w32process_module_get_filename (gpointer handle, gpointer module, gunichar2 *basename, guint32 size)
+gunichar2 *
+mono_w32process_module_get_filename (gpointer handle, gpointer module, guint32 *len)
 {
-	gint pid, len;
-	gsize bytes;
+	gint pid;
+	gsize bytes = 0;
 	gchar *path;
 	gunichar2 *proc_path;
 
-	size *= sizeof (gunichar2); /* adjust for unicode characters */
+	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER_PROCESS, "%s: Getting module file name, process handle %p module %p " G_GUINT32_FORMAT,
+	            __func__, handle, module);
 
-	if (basename == NULL || size == 0)
-		return 0;
+	*len = 0;
 
 	pid = mono_w32process_get_pid (handle);
+	if (pid == 0)
+		return NULL;
 
 	path = mono_w32process_get_path (pid);
 	if (path == NULL)
-		return 0;
+		return NULL;
 
 	proc_path = mono_unicode_from_external (path, &bytes);
-	g_free (path);
-
-	if (proc_path == NULL)
-		return 0;
 
-	len = (bytes / 2);
+	*len = bytes / sizeof (gunichar2);
```

Comment:
Aside: Windows code used to always `<< 1` and `>> 1` but that fell out of fashion.